### PR TITLE
Cliprect op

### PR DESCRIPTION
**Pull request by ferhatb**
## Description

- Part1 for supporting ClipOp option in Canvas.clipRect API. Propogated clipOp into PathCommand.
- Fix nullability in PathCommand.apply.

## Related Issues

https://github.com/flutter/flutter/issues/60006
https://github.com/flutter/flutter/issues/64734

## Tests

No change to tests for nullability. Once ClipOp is working will add 64734 issue sample as golden test.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [contributor guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [C++, Objective-C, Java style guides] for the engine.
- [x] I read the [tree hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation.
- [x] All existing and new tests are passing.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

### CODE REVIEW

File: lib/web_ui/lib/src/engine/html/recording_canvas.dart [hunk 1]
```diff
@@ -1610,9 +1627,9 @@ class _PaintBounds {
   double _bottom = -double.maxFinite;
 
   // Stack of transforms.
-  late List<Matrix4> _transforms = <Matrix4>[];
```
Comment (nturgut) [inline lib/web_ui/lib/src/engine/html/recording_canvas.dart:1613]: can this be final?

File: lib/web_ui/lib/src/engine/html/recording_canvas.dart [hunk 2]
```diff
@@ -1610,9 +1627,9 @@ class _PaintBounds {
...
+  List<Matrix4> _transforms = <Matrix4>[];
   // Stack of clip bounds.
-  late List<ui.Rect?> _clipStack = <ui.Rect?>[];
```
Comment (nturgut) [inline lib/web_ui/lib/src/engine/html/recording_canvas.dart:1615]: same for this one.

File: lib/web_ui/lib/src/engine/html/recording_canvas.dart (hunk 1 above)
Comment (ferhatb) [reply lib/web_ui/lib/src/engine/html/recording_canvas.dart:1613]: Thank you for catching this. Done.

File: lib/web_ui/lib/src/engine/html/recording_canvas.dart (hunk 2 above)
Comment (ferhatb) [reply lib/web_ui/lib/src/engine/html/recording_canvas.dart:1615]: Done

**Pull request merged**